cloud/docker.go

```diff
@@ -41,6 +44,34 @@ func (settings *dockerSettings) Validate() error {
 	return nil
 }
 
+func (s *dockerSettings) FromDistroSettings(d distro.Distro, _ string) error {
+	if len(d.ProviderSettingsList) != 0 {
+		bytes, err := d.ProviderSettingsList[0].MarshalBSON()
+		if err != nil {
+			return errors.Wrap(err, "error marshalling provider setting into bson")
+		}
+		if err := bson.Unmarshal(bytes, s); err != nil {
+			return errors.Wrap(err, "error unmarshalling bson into provider settings")
+		}
+	} else if d.ProviderSettings != nil {
+		if err := mapstructure.Decode(d.ProviderSettings, s); err != nil {
+			return errors.Wrapf(err, "Error decoding params for distro %s: %+v", d.Id, s)
+		}
+		bytes, err := bson.Marshal(s)
+		if err != nil {
+			return errors.Wrap(err, "error marshalling provider setting into bson")
+		}
+		if err := d.UpdateProviderSettings(bytes); err != nil {
```

Comment:
shouldn't we return an error here?

Reply:
I wasn't sure if we wanted this to impact our ability to spawn hosts, but probably this wouldn't happen often unless there's a bug in updating so sure.

Reply:
Actually, there some cases where we don't want to update or return an error (for example, when creating a new distro we first run the validator, and we shouldn't try to update a distro that hasn't been created yet). So either this should only error if it's not ResultsNotFound, or FromDistroSettings() could take a shouldUpdate boolean. I prefer the first option.